kghDataManagement/views.py

```diff
@@ -1,3 +1,167 @@
+                        change = self.handleItemPriceChange(row, kghItem)
+                        # If a change was made
+                        if (len(change) != 0):
+                            change[self.CHANGE_KGH_ID] = kghItem.kghID
+                            changes.append(change)
+
+                    # If an item matches an old KGH item ID
+                    elif (kghItem := self.getItemMatchingOldIds(row.get(self.ROW_OLD_MATERIAL), kghIdDict)) is not None:
+                        change = self.handleItemPriceChange(row, kghItem)
+
+                        title = kghItem.title
+                        newKghId = row.get(self.ROW_MATERIAL)
+
+                        change[self.CHANGE_TITLE] = title
+                        change[self.CHANGE_KGH_ID] = newKghId
+
+                        kghItem.kghID = newKghId
+
+                        changes.append(change)
+
+                Item.objects.bulk_update(list(kghIdDict.values()), ["kghID", "price"])
+        except IntegrityError as e:
+            messages.error(request, f"Error: {str(e)}")
+            traceback.print_exc()
+            return redirect('kgh-upload')
+        except Exception as e:
+            messages.error(request, f"Error: {str(e)}")
+            traceback.print_exc()
+            return redirect('kgh-upload')
+
+        messages.success(request, "File upload successful")
+        context = {
+            "changes": changes,
+            "catalogUploaded": True
+        }
+        #  TODO, make a table of all changes (context already provided)
+        # If no changes are made, let the user know
+        # Use the global keys to reference each field
+
+        return render(request, 'kghDataManagement/kgh_upload_page.html', context=context)
+
+    # row: A row (as a dictionary) of the KGH catalog, as parsed from decode_utf8()
+    # kghItem: An [Item] that has a valid KGH ID (either outdated or present)
+    # Updates kghItem price and returns a dictionary of the changes made for the response context
+    def handleItemPriceChange(self, row, kghItem):
+        newPriceString = row.get(self.ROW_PRICE).strip()
+        newPrice = StringUtils.getFloatOrNone(newPriceString)
+        if newPrice == None:
+            raise Exception(
+                f"Could not parse material price for item with KGH ID: {kghItem.kghID}. Please make sure it is a decimal with no non-digit characters.")
+        oldPrice = kghItem.price
+        if float(oldPrice) != newPrice:
+            kghItem.price = newPrice
+            return ({
+                self.CHANGE_TITLE: kghItem.title,
+                self.CHANGE_OLD_PRICE: oldPrice,
+                self.CHANGE_NEW_PRICE: newPrice,
+                self.CHANGE_OLD_KGH_ID: kghItem.kghID
+            })
+        return {}
+
+    # oldIds: A space separated string of old KGH IDs
+    # itemDictionary: itemDictionary[kghID] == Item
+    def getItemMatchingOldIds(self, oldIds, itemDictionary):
+        ids = oldIds.split(" ")
```

Comment:
Not always this simple, there are some edge cases
some of the nonstandard values you can find in old material no.

NO NUMBER ON FORM
NO NUMBER ON FORM
NO NUMBER ON FORM
NO NUMBER ON FORM
NO NUMBER ON FORM
SAP # 315
NO NUMBER ON FORM
SAP# 63576
FORM 43818
1.30390110235138E+017

Reply:
What is the best thing to do for these edge cases? 
NO NUMBER ON FORM: I'm going to just return nothing
SAP # 315: We should figure out what this means unless you know
FORM 43818: Not sure
1.30390110235138E+017: I guess that can be treated as an ID? I'm not sure

Reply:
You can treat the entire thing as an ID but the issue is there are internal spaces. I can't think of a great solution. maybe split on spaces, combine text with together until you hit a number and just have a special case for "NO NUMBER ON FORM"